src/python/Services/ReqMgr/ReqMgrReader.py

```diff
@@ -6,53 +6,383 @@
 
+        except Exception as error:
+            self.logger.error(
+                f"Failed to get workflows from reqmgr for dataset {dataset}"
+            )
+            self.logger.error(str(error))
+
+    def getWorkflowSchemaByPrepId(
+        self, pid: str, details: bool = False
+    ) -> Union[List[dict], List[str]]:
+        """
+        The function to get the list of workflows for a given prep id
+        :param pid: id
+        :param details: if True, it returns details for each workflow, o/w just workflows names
+        :return: a list of dicts if details is True, list of strings o/w
+        """
+        try:
+            return self._getWorkflowSchemaByParam({"prep_id": pid}, details)
+
+        except Exception as error:
+            self.logger.error(f"Failed to get workflows from reqmgr for id {pid}")
+            self.logger.error(str(error))
+
+    def getWorkflowsByStatus(
+        self,
+        status: str,
+        user: Optional[str] = None,
+        details: bool = False,
+        rtype: Optional[str] = None,
+        priority: Optional[str] = None,
+        tries: int = 5,
+        wait: int = 5,
+    ) -> Union[List[dict], List[str]]:
+        """
+        The function to get the list of workflows for a given status (and user/request type/priority)
+        :param status: workflow status
+        :param user: request user, if any
+        :param details: if True, it returns it returns details for each workflow, o/w just workflows names
+        :param type: request type, if any
+        :param priority: request priority, if any
+        :param tries: number of tries
+        :param wait: wait time between tries
+        :return: a list of dicts if details is True, list of strings o/w
+        """
+        try:
+            param = {"status": status}
+            if user:
+                param["requestor"] = user.split(",")
+            if rtype:
+                param["request_type"] = rtype
+            if priority:
+                # TODO: does this work? if not, can I remove this?
+                param["initialpriority"] = priority
+                self.logger.info(f"Priority {priority} is requested")
 
-            data = result['result']
-            if details:
-                ## list of dict
-                r = []
-                for it in data:
-                    r.extend(it.values())
-                return r
-            else:
+            return self._runWithRetries(
+                self._getWorkflowSchemaByParam,
+                [param],
+                {"details": details},
+                tries=tries,
+                wait=wait,
+            )
+
+        except Exception as error:
+            self.logger.error(
+                f"Failed to get workflows from reqmgr for {status} (user={user}, type={rtype}, priority={priority})"
+            )
+            self.logger.error(str(error))
+
+    def getWorkflowSchemaByName(
```

Comment:
Change name to`getWorkflowsByNames`. So that we understand that it gets multiple names and returns multiple workflows.